python/src/trezorlib/btc.py

```diff
@@ -294,32 +303,32 @@ def copy_tx_meta(tx: messages.TransactionType) -> messages.TransactionType:
 
         if res.request_type == R.TXMETA:
             msg = copy_tx_meta(current_tx)
-            res = client.call(messages.TxAck(tx=msg))
-
-        elif res.request_type in (R.TXINPUT, R.TXORIGINPUT):
+        elif res.request_type == R.TXEXTRADATA:
+            assert res.details.extra_data_offset is not None
+            assert res.details.extra_data_len is not None
+            assert current_tx.extra_data is not None
+            o, l = res.details.extra_data_offset, res.details.extra_data_len
+            msg = messages.TransactionType(extra_data=current_tx.extra_data[o : o + l])
+        else:
+            assert res.details.request_index is not None
```

Comment:
did you reflow this code just to have only one `assert res.details.request_index` ?

please don't do that and spell out the requirement in each branch.
(it's more correct that way; the requirement for `request_index` is not "for all others" but for each entry.
also the else-branch is semantically mismatched, the if/elif sequence is an equivalent of a case-statement where each `request_type` has an entry, so all of them should be on top level)

or is there another reason I'm not seeing?

Reply:
You are right, I made it so I can check that assert only once, will revert it and check that `request_index` before each usage. 

What about the creation of empty `msg` object, is it a good idea to create it in front of this whole `if/elif` (just the first `if res.request_type == R.TXMETA` will create its own )

Reply:
honestly, this might be a good opportunity to rewrite to the more specific `TxAck*` code:
every `request_type` has a corresponding `TxAck*` message type, which usually has a limited number of entries that match only that thing.

Reply:
Reverted in 3c5b9b6. Should we do the `TxAck*`thing as a part of this PR?